config/initializers/content_security_policy.rb

```diff
@@ -0,0 +1,61 @@
+require 'feature_management'
+
+if FeatureManagement.rails_csp_tooling_enabled?
+  # rubocop:disable Metrics/BlockLength
+  Rails.application.config.content_security_policy do |policy|
+    connect_src = ["'self'", '*.nr-data.net', '*.google-analytics.com', 'us.acas.acuant.net']
+
+    font_src = [:self, :data, IdentityConfig.store.asset_host.presence].compact
+
+    image_src = [
+      "'self'",
+      'data:',
+      'login.gov',
+      IdentityConfig.store.asset_host.presence,
+      'idscangoweb.acuant.com',
+      IdentityConfig.store.aws_region.presence &&
+        "https://s3.#{IdentityConfig.store.aws_region}.amazonaws.com",
+    ].select(&:present?)
+
+    script_src = [
+      :self,
+      'js-agent.newrelic.com',
+      '*.nr-data.net',
+      'dap.digitalgov.gov',
+      '*.google-analytics.com',
+      IdentityConfig.store.asset_host.presence,
+    ].compact
+
+    script_src = [:self, :unsafe_eval] if !Rails.env.production?
+
+    style_src = [:self, IdentityConfig.store.asset_host.presence].compact
+
+    if ENV['WEBPACK_PORT']
+      connect_src << "ws://localhost:#{ENV['WEBPACK_PORT']}"
+      script_src << "localhost:#{ENV['WEBPACK_PORT']}"
+    end
+
+    if IdentityConfig.store.rails_mailer_previews_enabled
+      # CSP 2.0 only; overriden by x_frame_options in some browsers
+      policy.frame_ancestors :self
+    end
```

Comment:
Maybe we should merge in these changes here?

https://github.com/18F/identity-idp/pull/5852/files

Reply:
Yeah, just saw that. I was gonna wait for those to get to main and pull it in.

Reply:
I got it for the new API in 2332f79c7ce2fad1247d02c65ce6b740cbcccd47. Once I merge your change in we should be ready to rock.

Reply:
Got it for the SecureHeaders API in c35cf3025ebeffc1c200d20bc5b5621a1ef82181